app/frontend/src/components/Navigation/Navigation.tsx

```diff
@@ -113,7 +95,34 @@ export default function Navigation() {
   const authenticated = useAuthContext().authState.authenticated;
   const [open, setOpen] = React.useState(false);
 
-  const drawerItems = (
+  const { data: user } = useCurrentUser();
+
+  const menu = useMemo(
+    () =>
+      user
+        ? [
+            {
+              name: "Dashboard",
+              route: "/",
+            },
+            {
+              name: "Messages",
+              route: messagesRoute,
+            },
+            {
+              name: "Search",
+              route: searchRoute,
+            },
+            {
+              name: "Profile",
+              route: routeToUser(user.username),
+            },
+          ]
+        : null,
```

Comment:
I'm having visions of the navigation buttons disappearing if the user is offline and their cache expires or something 😆 

Maybe a better solution is, if `user` is loading (undefined), to link to "/profile" and have that page just be a minimal widget that displays a loading indicator while `useCurrentUser()` is loading, then redirect to "/profile/username".